sentry_logging/lib/src/logging_integration.dart

```diff
@@ -0,0 +1,76 @@
+import 'dart:async';
+
+import 'package:logging/logging.dart';
+import 'package:sentry/sentry.dart';
+import 'version.dart';
+import 'extension.dart';
+
+/// An [Integration] which listens to all messages of the
+/// [logging](https://pub.dev/packages/logging) package.
+class LoggingIntegration extends Integration<SentryOptions> {
+  /// Creates the [LoggingIntegration].
+  ///
+  /// Setting [logExceptionAsEvent] to true (default) captures all
+  /// messages with errors as an [SentryEvent] instead of an [Breadcrumb].
+  /// Setting [logExceptionAsEvent] to false captures everything as
+  /// [Breadcrumb]s.
+  LoggingIntegration({
+    Level minBreadcrumbLevel = Level.INFO,
+    Level minEventLevel = Level.SEVERE,
+  })  : _minBreadcrumbLevel = minBreadcrumbLevel,
+        _minEventLevel = minEventLevel;
+
+  final Level _minBreadcrumbLevel;
+  final Level _minEventLevel;
+  late StreamSubscription<LogRecord> _subscription;
+  late Hub _hub;
+
+  @override
+  FutureOr<void> call(Hub hub, SentryOptions options) {
+    _hub = hub;
+    _setSdkVersion(options);
+    _subscription = Logger.root.onRecord.listen(
+      _onLog,
+      onError: (Object error, StackTrace stackTrace) {
+        _hub.captureException(error, stackTrace: stackTrace);
```

Comment:
you probably need to make it async/await

Reply:
I don't think we necessarily need async/await here, but I still changed it to use it.
If a future is not awaited, it's just a fire and forget operation. Because the closure is an `async void` method it is called as a fire and forget method anyways. I hope you get what I mean 😅 The same is true for your other comment.

Reply:
the problem is that if the exception or stack trace is null, we use `stacktrace.current`, if its not an awaited operation, the stack trace might change, also, if this is a log just before crashing an App, maybe there was no time to execute the operation, not sure if it'd be the case with Dart/Flutter but it is for multi-threaded languages